src/main/java/edu/brown/cs/student/main/Main.java

```diff
@@ -73,33 +76,175 @@ private void run() {
+                  // shuffles the equidistance stars in place to guarantee randomness, then add all to res array
+                  Collections.shuffle(sortedhm2.get(dist));
+                  res.addAll(sortedhm2.get(dist));
+                }
+              }
+              // print the first K starIDs
+              for (int i = 0; i < k; i++) {
+                System.out.println(res.get(i));
+              }
+            } catch (Exception e) {
+              System.out.println(
+                  "ERROR: We couldn't process your input, please check if you follow the format: naive_neighbors k x y z");
             }
-            else if (arguments[0].equalsIgnoreCase("subtract")) {
-              System.out.println(math.subtract(num1, num2));
+          }
+          // case where INPUT is in the format of naive_neighbors k "name"
+          else if (args.length >= 3 && args[0].equals("naive_neighbors") &&
+              args[2].contains("\"")) {
+            try {
+              //  create hashmap2 --> {distance: starIDs}
+              hm2 = new HashMap<>();
+              int k = Integer.parseInt(args[1]);
+              String starName;
+              // case where there's no empty space in the starName -> e.g. "Sol"
+              if (args.length == 3) {
+                starName = args[2].replace("\"", "");
+              } else { // case there's empty space, thus args is longer than 3 and the third element has "\"" -> e.g. "Lonely Star"
+                // extract name in form of list
+                ArrayList<String> starNameWithQuoteLst =
+                    new ArrayList<>(Arrays.asList(args).subList(2, args.length));
+
+                // join name by delimiter " " and get rid of "\""
+                String starNameWithQuote = String.join(" ", starNameWithQuoteLst);
+                starName = starNameWithQuote.replace("\"", "");
+
+              }
+              // case where the starName is not in the dataset
+              if (!this.hm1.containsKey(starName)) {
+                System.out.println("ERROR: The star couldn't be found");
+              } else {
+                Double[] starCoordinate = this.hm1.get(starName);
+                for (String[] star : stars) { // stars => ["1234", "awesomeStar", "21.00", "2398.23", "284.23"]
+                  Double[] coordinate = distCalculator.formCoordinate(star[2], star[3], star[4]);
+                  double distDiff = distCalculator.euclideanDistance(starCoordinate, coordinate);
+                  if (!hm2.containsKey(distDiff)) {
+                    hm2.put(distDiff, new ArrayList<>());
+                  }
+                  hm2.get(distDiff).add(star[0]);
+                }
+                // then sort the hashmap by key value (difference in dist), ascending
+                Map<Double, ArrayList<String>> sortedhm2 = new TreeMap<>(hm2);
+                ArrayList<String> res = new ArrayList<>();
+                for (Double dist : sortedhm2.keySet()) {
```

Comment:
Helper functions, where possible, may have been useful to make the code more readable